cmake/luajit.cmake

```diff
@@ -39,6 +39,8 @@ set(LUAJIT_SMART_STRINGS ON CACHE BOOL
     "Harder string hashing function" FORCE)
 set(LUAJIT_TEST_BINARY $<TARGET_FILE:tarantool> CACHE STRING
     "Lua implementation to be used for tests (tarantool)" FORCE)
+set(LUAJIT_TEST_DEPS "tarantool" CACHE STRING
```

Comment:
> Q: Do we need this line added since we use TARGET_FILE:tarantool above?

`LUAJIT_TEST_BINARY` and `LUAJIT_TEST_DEPS` have different meaning. The first one is an executable used for running tests, the second one is a number of dependencies used for running tests. These variables set to different values be default, see https://github.com/tarantool/luajit/blob/1e77778b0d413cbffeaffdece1f3aa474a686f16/CMakeLists.txt#L417-L424

Also, `LUAJIT_TEST_DEPS` is used in dependences, see
```
third_party/luajit/CMakeLists.txt:set(LUAJIT_TEST_DEPS "luajit-main" CACHE STRING
third_party/luajit/test/PUC-Rio-Lua-5.1-tests/CMakeLists.txt:  DEPENDS ${LUAJIT_TEST_DEPS} PUC-Rio-Lua-5.1-tests-prepare
third_party/luajit/test/tarantool-tests/CMakeLists.txt:  DEPENDS ${LUAJIT_TEST_DEPS} tarantool-tests-libs
third_party/luajit/test/tarantool-c-tests/CMakeLists.txt:  # XXX: LUAJIT_TEST_DEPS is not needed, we know dependencies
third_party/luajit/test/lua-Harness-tests/CMakeLists.txt:  DEPENDS ${LUAJIT_TEST_DEPS}
third_party/luajit/test/LuaJIT-tests/CMakeLists.txt:  DEPENDS ${LUAJIT_TEST_DEPS} LuaJIT-tests-libs
```
while `LUAJIT_TEST_BINARY` is not:

```
third_party/luajit/CMakeLists.txt:# See the rationale below (near LUAJIT_TEST_BINARY variable).
third_party/luajit/CMakeLists.txt:set(LUAJIT_TEST_BINARY ${LUAJIT_BINARY} CACHE STRING
third_party/luajit/test/tarantool-tests/lj-551-bytecode-c-broken-macro/CMakeLists.txt:      LUA_PATH=${LUA_PATH} ${LUAJIT_TEST_BINARY} -b ${LUA_SOURCE} ${EXT_SOURCE}
third_party/luajit/test/tarantool-tests/lj-551-bytecode-c-broken-macro/CMakeLists.txt:    DEPENDS ${LUAJIT_TEST_BINARY} ${LUA_FILE}
third_party/luajit/test/tarantool-tests/lj-549-bytecode-loader/CMakeLists.txt:    LUA_PATH=${LUA_PATH} ${LUAJIT_TEST_BINARY} -b ${LUA_FILE} ${C_FILE}
third_party/luajit/test/tarantool-tests/lj-549-bytecode-loader/CMakeLists.txt:  DEPENDS ${LUAJIT_TEST_BINARY} ${LUA_FILE}
third_party/luajit/test/CMakeLists.txt:set(LUAJIT_TEST_COMMAND "${LUAJIT_TEST_BINARY} -e dofile[[${LUAJIT_TEST_INIT}]]")
```

>  If yes, why is this placed in this commit instead of the separate one?

I'll put it to a separate commit.